test/cri-containerd/logging_binary_test.go

```diff
@@ -0,0 +1,144 @@
+				runtimeHandler:   wcowProcessRuntimeHandler,
+				sandboxImage:     imageWindowsNanoserver,
+				containerImage:   imageWindowsNanoserver,
+				cmd:              []string{"ping", "-t", "127.0.0.1"},
+			},
+			expectedContent: "Pinging 127.0.0.1 with 32 bytes of data",
+		},
+		{
+			TestConfig: TestConfig{
+				name:             "WCOW_Hypervisor",
+				containerName:    t.Name() + "-Container-WCOW_Hypervisor",
+				requiredFeatures: []string{featureWCOWHypervisor},
+				runtimeHandler:   wcowHypervisorRuntimeHandler,
+				sandboxImage:     imageWindowsNanoserver,
+				containerImage:   imageWindowsNanoserver,
+				cmd:              []string{"ping", "-t", "127.0.0.1"},
+			},
+			expectedContent: "Pinging 127.0.0.1 with 32 bytes of data",
+		},
+		{
+			TestConfig: TestConfig{
+				name:             "LCOW",
+				containerName:    t.Name() + "-Container-LCOW",
+				requiredFeatures: []string{featureLCOW},
+				runtimeHandler:   lcowRuntimeHandler,
+				sandboxImage:     imageLcowK8sPause,
+				containerImage:   imageLcowAlpine,
+				cmd:              []string{"ash", "-c", "while true; do echo 'Hello, World!'; sleep 1; done"},
+			},
+			expectedContent: "Hello, World!",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			requireFeatures(t, test.requiredFeatures...)
+
+			requiredImages := []string{test.sandboxImage, test.containerImage}
+			if test.runtimeHandler == lcowRuntimeHandler {
+				pullRequiredLcowImages(t, requiredImages)
+			} else {
+				pullRequiredImages(t, requiredImages)
+			}
+
+			podReq := getRunPodSandboxRequest(t, test.runtimeHandler)
+			podID := runPodSandbox(t, client, ctx, &podReq)
+			defer removePodSandbox(t, client, ctx, podID)
+
+			logFileName := fmt.Sprintf("%s/stdout-%s.txt", logBinaryRoot, test.name)
+			conReq := getCreateContainerRequest(podID, test.containerName, test.containerImage, test.cmd, podReq.Config)
+			conReq.Config.LogPath = logPath + fmt.Sprintf("?%s", logFileName)
+
+			createAndRunContainer(t, client, ctx, conReq)
+
+			_, err := os.Stat(logFileName)
+
+			if os.IsNotExist(err) {
+				t.Fatalf("log file was not created: %s", logFileName)
+			}
+			defer os.Remove(logFileName)
+
+			ok, err := assertFileContent(logFileName, test.expectedContent)
+
+			if err != nil {
+				t.Fatalf("failed to read log file: %s", err)
+			}
+
+			if !ok {
+				t.Fatalf("file content validation failed: %s", test.expectedContent)
+			}
+		})
+	}
+}
+
+func createAndRunContainer(t *testing.T, client runtime.RuntimeServiceClient, ctx context.Context, conReq *runtime.CreateContainerRequest) {
+	containerID := createContainer(t, client, ctx, conReq)
+	defer removeContainer(t, client, ctx, containerID)
+
+	startContainer(t, client, ctx, containerID)
+	defer stopContainer(t, client, ctx, containerID)
+
+	// Let stdio kick in
+	time.Sleep(time.Second * 1)
+}
+
+func assertFileContent(path string, content string) (bool, error) {
+	fileContent, err := ioutil.ReadFile(path)
+
```

Comment:
Can we move the error checking up one line? Generally it's the line after we receive the error in go